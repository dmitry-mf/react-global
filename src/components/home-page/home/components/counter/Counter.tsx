import React from "react";
import cn from "classnames/bind";
import styles from '../../../../../styles/components/index.module.scss';
const cx = cn.bind(styles);

export const Counter: React.FC<{count: number | string}> = ({count}) => {

    const counterCn = cx(
        'counter',
    );

    const countCn = cx(
        'text',
        'text_lg',
        'text_white',
        'text_bold',
        'padding-h_right_10px',
    );

    const descriptionCn = cx(
        'text',
        'text_normal',
        'text_md',
        'text_white',
    )

    return (
        <div className={counterCn}>
            <span className={countCn}>{count}</span>
            <span className={descriptionCn}>movies found</span>
        </div>
    )
}
