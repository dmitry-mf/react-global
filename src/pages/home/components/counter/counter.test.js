import '@testing-library/jest-dom'
import * as React from 'react';

import { render } from '@testing-library/react';
import { Counter } from './Counter';

it('render counter component', () => {
    const { asFragment } = render(<Counter count={2}/>);
    expect(asFragment()).toMatchSnapshot();
});