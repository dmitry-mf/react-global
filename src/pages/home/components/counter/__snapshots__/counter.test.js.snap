// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render counter component 1`] = `
<DocumentFragment>
  <div
    class="counter"
  >
    <span
      class="text text_lg text_white text_bold padding-h_right_10px"
    >
      2
    </span>
    <span
      class="text text_normal text_md text_white"
    >
      movies found
    </span>
  </div>
</DocumentFragment>
`;
